src/core/Akka/Util/ByteString.cs

```diff
@@ -459,53 +382,38 @@ public int IndexOf(byte b, int from)
         /// <returns></returns>
         public bool HasSubstring(ByteString other, int index)
         {
-            // quick check: if subsequence is longer than remaining size, return false
-            if (other.Count > _count - index) return false;
+            if (other.Count == 0) return true; // Empty spans are always "found".
```

Comment:
per discord comments, if we can use the 'Contains' extensions that exist in `System.Memory`, namely `public static int IndexOf<T>(this Span<T> span, ReadOnlySpan<T> value) where T : IEquatable<T>`, it would be strongly encouraged.